storm-client/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -36,10 +37,21 @@
     private final MetricRegistry registry = new MetricRegistry();
     private final List<StormReporter> reporters = new ArrayList<>();
     private String hostName = null;
+    private boolean useShortName = false;
+
+    public StormMetricRegistry(boolean useShortName) {
```

Comment:
I would pass the topology conf instead in case we want to add other future options to configure this class.